### PR TITLE
Quote column name

### PR DESCRIPTION
**Pull request by suzaku**

### CODE REVIEW

**Comment by kennytm** [comment]
What if the column name contains a backquote like ``foo`bar``?

**Comment by suzaku** [comment]
What's the correct way to escape "`"s in column name?

**Comment by kennytm** [comment]
@suzaku the quoted form becomes ``` `foo``bar` ```.

Use [mysql_real_escape_string_quote](https://dev.mysql.com/doc/refman/8.0/en/mysql-real-escape-string-quote.html) with ``'`'`` as final argument.

**Comment by csuzhangxc** [comment]
> @suzaku the quoted form becomes `` `foo``bar` ``.
> 
> Use [mysql_real_escape_string_quote](https://dev.mysql.com/doc/refman/8.0/en/mysql-real-escape-string-quote.html) with `` '`' `` as final argument.

@kennytm it cannot be used to escape `"` with `ANSI_QUOTES` SQL mode?

> If the ANSI_QUOTES SQL mode is enabled, mysql_real_escape_string_quote() cannot be used to escape double quote characters for use within double-quoted identifiers. (The function cannot tell whether the mode is enabled to determine the proper escaping character.)

**Comment by kennytm** [comment]
@csuzhangxc You always use `` ` `` to quote identifiers and always use `'` to quote strings, regardless of ANSI_QUOTES settings.

**Comment by csuzhangxc** [comment]
@kennytm PTAL

**Comment by kennytm** [comment]
`mysql_real_escape_string_quote` does not work?

**Comment by csuzhangxc** [comment]
> `mysql_real_escape_string_quote` does not work?

I try to quote (`desc` without ``` ` ```, but `mdb`/`mtbl` with ``` ` ```)

```
SELECT /*!40001 SQL_NO_CACHE */ c1,desc FROM `mdb`.`mtbl`
```

it returns (`desc` still without ``` ` ```, but `mdb`/`mtbl` with ``` `` ```)

```
SELECT /*!40001 SQL_NO_CACHE */ c1,desc FROM ``mdb``.``mtbl``
```

if try to quote `desc` (not a whole statement), it will just return `desc`.

**Comment by csuzhangxc** [comment]
@kennytm abstracted in https://github.com/pingcap/mydumper/pull/28/commits/dc2dd5b4c89e50cc10ba6f6d1b7f0882902819d3. PTAL again.

**Comment by csuzhangxc** [comment]
@amyangfei PTAL

File: mydumper.c [hunk 1]
```diff
@@ -1875,7 +1888,9 @@ GString * get_insertable_fields(MYSQL *conn, char *database, char *table){
 			g_string_append(field_list, ",");
 		}
 
-		g_string_append_len(field_list, bind.buffer, *bind.length);
+		GString *original = g_string_new_len(bind.buffer, *bind.length);
+        append_escaped_identifier(field_list, original->str);
```
Comment (amyangfei) [inline mydumper.c]: most code in this repo uses `tab` as indent, better to be unified

Comment (csuzhangxc) [reply mydumper.c]: done in https://github.com/pingcap/mydumper/pull/28/commits/520f0789472d0e0fb72405e29bb8810e5e0c0884.

**Pull request merged**